Кроткая Александра, Задача по тестированию

Krotkaya opened the pull request:
@Zuguki

Krotkaya committed:
Домашка по тестированию, часть 1

ДЗ включало в себя:
- Использование FluentAssertions при написании тестов

Zuguki commented on Testing/Basic/Homework/1. ObjectComparison/ObjectComparison.cs:
```diff
@@ -10,34 +12,52 @@ public class ObjectComparison
             new Person("Vasili III of Russia", 28, 170, 60, null));
-
+    
         // Перепишите код на использование Fluent Assertions.
```
Нужно переписать все проверки на Fluent Assertions

Krotkaya replied:
В этом классе ObjectComparison у меня только изначальный код для задания был написан через ClassicAssert (мое решение на Fluent Assertions было под ним снизу). 
Сейчас я убрала изначальный вариант первого теста полностью, теперь ClassicAssert есть только во втором тесте для того, чтобы наглядно видеть недостатки такой структуры.

Zuguki commented on Testing/Basic/Homework/2. NumberValidator/NumberValidatorTests.cs:
```diff
@@ -28,4 +28,61 @@ public void Test()
+    [Test]
+    public void Constructor_ThrowExceptions_InvalidParameter()
+    {
+        Assert.Throws<ArgumentException>(() => new NumberValidator(-1, 2, true));
```
давай в тестах на этот блок тоже заиспользуем FluentAssertions

Krotkaya replied:
NumberValidatorTests полностью переписала на FluentAssertions

Zuguki commented on Testing/Basic/Homework/2. NumberValidator/NumberValidatorTests.cs:
```diff
@@ -28,4 +28,61 @@ public void Test()
+    }
+    
+    [Test]
+    public void Constructor_DoesNotThrowExceptions_CorrectParameter()
```
Этот и тест выше, лучше вынести в testcases, иначе можем упасть на 1й проверке и будет не выполнен критерий: `при падении теста было без стек-трейса понятно на каких данных код не работает`

По хорошему писать тесты максимально атомарными

Krotkaya replied:
Да, понимаю почему, переделала

Zuguki reviewed (changes requested):
Посмотрел, нужно поправить

Krotkaya committed:
Внесла правки

Изменения:
- Убрала лишний код с заданием на ClassicAssert
- NumberValidatorTests добавила FluentAssertions
- Вынесла проверки на exceptions в TestCase
